src/utilities/helpers.ts

```diff
@@ -16,3 +24,31 @@ export const formatTime = (seconds: number) => {
   let secondsFormatted = seconds % 60 > 9 ? seconds % 60 : "0" + (seconds % 60);
   return hoursFormatted + minutesFormatted + secondsFormatted;
 };
+
+const setStreak = async (count: number) => {
+  await AsyncStorage.setItem(
+    "@streak",
+    JSON.stringify({ count, timestamp: moment().toISOString() }),
+  );
+  return count;
+};
+
+export const updateStreak = async (): Promise<number> => {
+  const streakData = await AsyncStorage.getItem("@streak").catch((e) =>
+    console.log(e),
+  );
+  if (streakData) {
+    const { count, timestamp } = JSON.parse(streakData) as Streak;
+    const lastMeditated = moment(timestamp);
+    const today = moment();
+    if (
+      !today.isSame(lastMeditated, "day") &&
+      today.diff(lastMeditated, "hours") < 24
```

Comment:
I don't think this will work, as we are saying "is the day not the same as today" and "is the time less that 24 hours ago"

If we meditated at 7pm one day, then 8pm the next day, the second condition is not met, when in reality the streak should continue.

Reply:
Oh so it's even easier then! We just need to check if `today.diff(lastMeditated, "days") === 1`

Reply:
Hmm, is that checked that the day part of the date (e.g. for a date 30/12/2019, the "30" part) is different by one, or is it it converting the entire date/time just to days since 1970?

I suspect this is the former, so if you meditated on 1st November, then on 2nd October, the streak would continue.

However, I don't understand momentjs, so I could be wrong here.

Reply:
yeah it checks the entire date